doctest.h

```diff
@@ -0,0 +1,5971 @@
+// ====================================================================== lgtm [cpp/missing-header-guard]
+// == DO NOT MODIFY THIS FILE BY HAND - IT IS AUTO GENERATED BY CMAKE! ==
+// ======================================================================
+//
+// doctest.h - the lightest feature-rich C++ single-header testing framework for unit tests and TDD
+//
+// Copyright (c) 2016-2019 Viktor Kirilov
+//
+// Distributed under the MIT Software License
+// See accompanying file LICENSE.txt or copy at
+// https://opensource.org/licenses/MIT
+//
+// The documentation can be found at the library's page:
+// https://github.com/onqtam/doctest/blob/master/doc/markdown/readme.md
```

Comment:
If we are using Git submodules for BLT, why would we not to the same for doctest?

Reply:
That was actually a mistake, we don’t have anything using doctest at the moment so I’ll probably rebase that out. The direct answer to the question though is that it being a single header vendoring it directly seems reasonable, where the convention for blt is normally to do a sub module. Honestly wouldn’t mind directly vendoring that either come to think of it though. Do you have a strong preference for one way vs the other?